Firestore/Source/Local/FSTLRUGarbageCollector.mm

```diff
@@ -0,0 +1,100 @@
+#import "Firestore/Source/Local/FSTLRUGarbageCollector.h"
+
+#import <queue>
+
+#import "Firestore/Source/Local/FSTMutationQueue.h"
+#import "Firestore/Source/Local/FSTQueryCache.h"
+
+const FSTListenSequenceNumber kFSTListenSequenceNumberInvalid = -1;
+
+class RollingSequenceNumberBuffer {
+ public:
+  RollingSequenceNumberBuffer(NSUInteger max_elements)
+      : max_elements_(max_elements), queue_(std::priority_queue<FSTListenSequenceNumber>()) {
+  }
+
+  RollingSequenceNumberBuffer(const RollingSequenceNumberBuffer &other) = delete;
```

Comment:
If you declare any copy constructor/assignment that implicitly precludes the move operations from being generated. The deletions of the moves are redundant.

https://stackoverflow.com/questions/33096653/make-a-class-non-copyable-and-non-movable